File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> mp;
        int l = 0, ans = 0;
        for(int i = 0;i<s.size(); i++){
            while(mp.find(s[i])!=mp.end()) {
                mp.erase(s[l]);
                l++;
            }
            mp.insert(s[i]);
            ans = max(ans,(int) mp.size());
        }
        return ans;
    }
};

File: Data Structures & Algorithms/insert-new-interval/submission-2.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        int n = intervals.length;
        int i = 0;

        List<int[]> ans = new ArrayList<>();

        while(i < n && intervals[i][1]<newInterval[0]) {
            ans.add(intervals[i]);
            i++;
        }

        while(i < n && newInterval[1] >= intervals[i][0]) {
            newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
            newInterval[1] = Math.max(newInterval[1], intervals[i][1]);
            i++;
        }
        ans.add(newInterval);

        while(i<n) {
            ans.add(intervals[i++]);
        }

        return ans.toArray(new int[ans.size()][]);
    }
}
